File: doc/src/snippets/qprocess/qprocess-simpleexecution.cpp
#include <QApplication>
#include <QtGui>

int main(int argc, char *argv[])
{
    QApplication app(argc, argv);
//! [0]
    QObject *parent;
//! [0]
    parent = &app;

//! [1]
    QString program = "./path/to/Qt/examples/widgets/analogclock";
//! [1]
    program = "./../../../../examples/widgets/analogclock/analogclock";

//! [2]
    QStringList arguments;
    arguments << "-style" << "motif";

    QProcess *myProcess = new QProcess(parent);
    myProcess->start(program, arguments);
//! [2]

    return app.exec();
}

File: doc/src/snippets/qprocess/qprocess-simpleexecution.java
import com.trolltech.qt.*;
import com.trolltech.qt.core.*;
import com.trolltech.qt.gui.*;

import java.util.*;

class SimpleExecution
{

    public static void main(String args[])
    {
        QApplication.initialize(args);
//! [0]
        QObject parent = QApplication.instance();
//! [0]

//! [1]
        String program = "./path/to/my/favorite/program";
//! [1]
        program = "./../../../../examples/widgets/analogclock/analogclock";

//! [2]
        List<String> arguments = new Vector<String>();
        arguments.add("-style");
        arguments.add("motif");

        QProcess myProcess = new QProcess(parent);
        myProcess.start(program, arguments);
//! [2]

        QApplication.execStatic();
        QApplication.shutdown();
    }
}
